### PR TITLE
Enable Windows on ARM builds for 0.16.x

### PR DESCRIPTION
**Pull request by svenpaulsen**
Here's a PR to enable building 0.16.20 for Windows on ARM as requested in #1551.

**Prerequisites to build from source:**

1. Download and install [LLVM](https://github.com/llvm/llvm-project/releases)
2. Download and install [Perl](https://strawberryperl.com)
3. Download and install [NASM](https://www.nasm.us/pub/nasm/releasebuilds/?C=M;O=D)

### CODE REVIEW

**Comment by awakecoding** [comment]
@svenpaulsen @briansmith I switched to this branch for my own builds, and it works 👍

**Comment by Alovchin91** [comment]
Hi @briansmith  Do you think this could be merged soon? It currently blocks quite a few projects…

File: src/cpu.rs [hunk 1]
```diff
@@ -238,6 +269,12 @@ pub(crate) mod arm {
 
             #[cfg(all(
                 any(target_os = "android", target_os = "fuchsia", target_os = "linux"),
```
Comment (roblabla) [inline src/cpu.rs:271]: Shouldn't this line get deleted? The new block `any` block right under contains all the same oses, + windows.

**Comment by briansmith** [comment]
This is a very big change and I could use some hand-holding to understand it. I suspect that this PR is mostly copying things from the existing main branch. It would be extremely helpful if you could go through chunk-by-chunk in the GitHub review interface and explain where you copied the change from and why that change was copied, and go into more depth about how you came up with this set of changes in the first place.

Especially regarding the PerlAsm changes, it would really help me if you could document why and from where you copied those changes.

**Comment by briansmith** [comment]
I left some comments in https://github.com/rust-lang/rustup/issues/2612#issuecomment-1433892701 that are very relevant to the development and review of this PR.

**Comment by complexspaces** [comment]
Hey everyone,

In case it is helpful, I just wanted to point out that we (1Password) backported the necessary components for WoA support in the 0.16 branch as part of https://github.com/1Password/ring/pull/8. The `build.rs` changes ended up being smaller then what was done here as the changes were rewritten to fit the state of `0.16` instead of backporting _all_ of 0.17's new design. We've tested it and AFAWCT, it functions as expected in cross-compilation and in an ARM64 VM.

**Comment by Alovchin91** [comment]
@briansmith Would it make it easier / less uncertain for you if someone (me?) created a pull request from the 1Password’s branch? I understand that you’re concerned about the amount of changes in this PR.

**Comment by Alovchin91** [comment]
@complexspaces Is it okay if I (or maybe you, if you’d prefer that) create a new PR from the branch you’ve mentioned?

**Comment by Alovchin91** [comment]
Alternatively, @svenpaulsen would you maybe kindly respond to @briansmith’s concerns here: https://github.com/briansmith/ring/pull/1554#issuecomment-1433885856 😊 I hope either should help.

**Comment by jeremyd2019** [comment]
> Especially regarding the PerlAsm changes, it would really help me if you could document why and from where you copied those changes.

https://github.com/openssl/openssl/commit/b863e1e4c69068e4166bdfbbf9f04bb07991dd40 which added Windows on ARM support to openssl

**Comment by briansmith** [comment]
Closing this as not planned. It has been years since we've done anything with 0.16.x. If any of these changes would be helpful for 0.17, PLMK. Thanks!

**Pull request closed without merging**